src/main/java/com/code_intelligence/jazzer/mutation/support/ParameterizedTypeSupport.java

```diff
@@ -0,0 +1,492 @@
+    return new AnnotatedWildcardTypeWrapper(annotated, resolvedType, resolvedLower, resolvedUpper);
+  }
+
+  private static class WildcardTypeWrapper implements WildcardType {
+    private final Type[] lowerBounds;
+    private final Type[] upperBounds;
+
+    public WildcardTypeWrapper(Type[] lowerBounds, Type[] upperBounds) {
+      this.lowerBounds = lowerBounds.clone();
+      this.upperBounds = upperBounds.clone();
+    }
+
+    @Override
+    public Type[] getUpperBounds() {
+      return upperBounds.clone();
+    }
+
+    @Override
+    public Type[] getLowerBounds() {
+      return lowerBounds.clone();
+    }
+
+    @Override
+    public String toString() {
+      Type[] lowerBounds = getLowerBounds();
+      Type[] bounds = lowerBounds;
+      StringBuilder sb = new StringBuilder();
+
+      if (lowerBounds.length > 0) sb.append("? super ");
+      else {
+        Type[] upperBounds = getUpperBounds();
+        if (upperBounds.length > 0 && !upperBounds[0].equals(Object.class)) {
+          bounds = upperBounds;
+          sb.append("? extends ");
+        } else return "?";
+      }
+
+      StringJoiner sj = new StringJoiner(" & ");
+      for (Type bound : bounds) {
+        sj.add(bound.getTypeName());
+      }
+      sb.append(sj);
+
+      return sb.toString();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+      if (!(other instanceof WildcardType)) {
+        return false;
+      }
+      WildcardType that = (WildcardType) other;
+      return Arrays.equals(getLowerBounds(), that.getLowerBounds())
+          && Arrays.equals(getUpperBounds(), that.getUpperBounds());
+    }
+
+    @Override
+    public int hashCode() {
+      return Arrays.hashCode(getLowerBounds()) * 31 + Arrays.hashCode(getUpperBounds());
+    }
+  }
+
+  private static class GenericArrayTypeWrapper implements GenericArrayType {
+
+    private final Type componentType;
+
+    public GenericArrayTypeWrapper(Type componentType) {
+      this.componentType = componentType;
+    }
+
+    @Override
+    public Type getGenericComponentType() {
+      return componentType;
+    }
+
+    @Override
+    public String toString() {
+      return componentType.getTypeName() + "[]";
+    }
+
+    @Override
+    public boolean equals(Object other) {
+      if (!(other instanceof GenericArrayType)) {
+        return false;
+      }
+      GenericArrayType that = (GenericArrayType) other;
+      return componentType.equals(that.getGenericComponentType());
+    }
+
+    @Override
+    public int hashCode() {
+      return componentType.hashCode();
+    }
+  }
+
+  private static class ParameterizedTypeWrapper implements ParameterizedType {
+    private final Type rawType;
+    private final Type[] typeArguments;
+    private final Type ownerType;
+
+    public ParameterizedTypeWrapper(Type rawType, Type[] typeArguments, Type ownerType) {
+      this.rawType = rawType;
+      this.typeArguments = typeArguments;
```

Comment:
Just making sure that callers cannot manipulate the internal state. This applies to the constructors like here, and also the array getters returning `AnnotatedType[]`. You did this for `WildcardTypeWrapper` but not for the others. I marked some more places below with missing `.clone()`
```suggestion
      this.typeArguments = typeArguments.clone();
```